src/main/java/at/hannibal2/skyhanni/test/TestCopyChestData.kt

```diff
@@ -0,0 +1,82 @@
+            copyChestData(chest = (event.gui as GuiChest).inventorySlots.inventorySlots)
+            return
+        } else if (config.copyPlayerInventory.isKeyHeld()) {
+            copyInventoryData(inventory = if (config.includeArmor) mc.thePlayer.inventory.mainInventory + mc.thePlayer.inventory.armorInventory else mc.thePlayer.inventory.mainInventory)
+            return
+        } else if (InventoryUtils.openInventoryName().isNotEmpty()) {
+            OSUtils.copyToClipboard(InventoryUtils.openInventoryName())
+            ChatUtils.chat("Chest name copied to clipboard.")
+            return
+        }
+    }
+    private fun copyInventoryData(inventory: Array<ItemStack?>) {
+        val copyList = mutableListOf<String>("relevant config:", "includeNullSlots: ${config.includeNullSlots}", "includeUnnamedItems: ${config.includeUnnamedItems}", "includeArmor: ${config.includeArmor}", "", "your inventory is below.", "", "your hotbar:", "")
+        for ((i, stack) in inventory.withIndex()) {
+            if (i == HOTBAR_SIZE) copyList.addAll(listOf("note: hotbar ends here", "the rest of your inventory:", "", ""))
+            if (i == HOTBAR_SIZE + INVENTORY_SIZE && config.includeArmor) copyList.addAll(listOf("note: the rest of your inventory data ends here", "your armor:", "", ""))
+            if (stack == null) {
+                if (config.includeNullSlots) copyList.addAll(listOf("(there is nothing inside slot $i; it is null)", "", ""))
+                continue
+            }
+            if (stack.displayName.isNotBlank() || config.includeUnnamedItems) {
+                copyList.add("slot index: '$i'")
+                copyList.addAll(stack.getStackInfo())
+            }
+        }
+        OSUtils.copyToClipboard(copyList.joinToString("\n"))
+        ChatUtils.chat("Inventory $DEBUG_SUFFIX")
+    }
+    private fun copyChestData(chest: List<Slot>) {
+        val copyList = mutableListOf<String>("relevant config:", "includeNullSlots: ${config.includeNullSlots}", "includeUnnamedItems: ${config.includeUnnamedItems}", "includeArmor: ${config.includeArmor}", "", "chest name: '${InventoryUtils.openInventoryName()}'", "")
+        copyList.addAll(listOf<String>())
+        for (slot in chest) {
+            val stack = slot.stack
+            if (stack == null) {
+                if (config.includeNullSlots) copyList.addAll(listOf("(there is nothing inside slot ${slot.slotIndex}; it is null)", "", ""))
+                continue
+            }
+            if (stack in mc.thePlayer.inventory.mainInventory) break
+            if ((stack.displayName.isNotEmpty() && stack.displayName.isNotBlank()) || config.includeUnnamedItems) {
+                copyList.add("slot index: '${slot.slotIndex}'")
+                copyList.addAll(stack.getStackInfo())
+            }
+        }
+        OSUtils.copyToClipboard(copyList.joinToString("\n"))
+        ChatUtils.chat("Chest $DEBUG_SUFFIX")
+    }
+    private fun ItemStack.getStackInfo(): List<String> {
```

Comment:
make the format more json like, eg. brackets. Not necessarily but makes it more readable in my opinion. Also does not need to be a full json.